pyccel/codegen/printing/ccode.py

```diff
@@ -2084,12 +2084,22 @@ def _print_NumpySign(self, expr):
         self.add_import(c_imports['pyc_math_c'])
         primitive_type = expr.dtype.primitive_type
         func = ''
+        arg = self._print(expr.args[0])
+
         if isinstance(primitive_type, PrimitiveIntegerType):
             func = 'isign'
         elif isinstance(primitive_type, PrimitiveFloatingPointType):
             func = 'fsign'
         elif isinstance(primitive_type, PrimitiveComplexType):
-            func = 'csign'
+            if expr.dtype.precision == 4:
+                return f'py_sign_type_float_complex({arg})'
+            elif expr.dtype.precision == 8:
+                return f'py_sign_type_double_complex({arg})'
+            elif expr.dtype.precision == 16:
+                return f'py_sign_type_long_double_complex({arg})'
+            else:
+                raise ValueError(f"Unsupported complex precision: {expr.dtype.precision}")
```

Comment:
I still haven't written the documentation on this but Pyccel has its own error handling system so that errors for users are printed without a scary traceback and including the file and line number where the error was found. To use this system you should write:

```suggestion
                errors.report(f"Unsupported complex precision: {expr.dtype.precision}",
                     severity='fatal', symbol=expr)
```
`fatal` means that printing cannot continue after this error. It is also possible to choose a default value for the function name and use `severity='error'`. In this case printing will continue and the error will be raised when Pyccel has finished the printing stage. This allows multiple errors to be raised for users before the translation quits.